core/src/main/java/org/gbif/pipelines/core/functions/transforms/ExtendedRecordToEventTransformer.java

```diff
@@ -0,0 +1,123 @@
+package org.gbif.pipelines.core.functions.transforms;
+
+import org.gbif.dwca.avro.Event;
+import org.gbif.pipelines.core.functions.interpretation.DayInterpreter;
+import org.gbif.pipelines.core.functions.interpretation.InterpretationException;
+import org.gbif.pipelines.core.functions.interpretation.MonthInterpreter;
+import org.gbif.pipelines.core.functions.interpretation.YearInterpreter;
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueLineageRecord;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+import org.gbif.pipelines.io.avro.ExtendedRecord;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.TupleTag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This function converts an extended record to an interpreted KeyValue of occurenceId and Event.
+ * This function returns multiple outputs,
+ * a. Interpreted version of raw temporal data as KV<String,Event>
+ * b. Issues and lineages applied on raw data to get the interpreted result, as KV<String,IssueLineageRecord>
+ */
+public class ExtendedRecordToEventTransformer extends DoFn<ExtendedRecord, KV<String, Event>> {
+
+  /**
+   * tags for locating different types of outputs send by this function
+   */
+  public static final TupleTag<KV<String, Event>> eventDataTag = new TupleTag<KV<String, Event>>();
```

Comment:
declarations like public static final TupleTag<KV<String, Event>> eventDataTag = new TupleTag<KV<String, Event>>();  can be transformed to public static final TupleTag<KV<String, Event>> eventDataTag = new TupleTag<>();  see https://docs.oracle.com/javase/tutorial/java/generics/restrictions.html